### ISSUE TITLE
Double quotes in alt text are not escaped

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctor#2061]
Double quotes in alt text are not escaped when generating HTML and DocBook. This, in turn, produces a document which does not validate.

Example:

```
image::scream.jpg[The painting "The Scream" by Edvard Munch]
```

Result:

```
<div class="imageblock">
<div class="content">
<img src="scream.jpg" alt="The painting "The Scream" by Edvard Munch">
</div>
</div>
```

This also affects inline images.

**Comment by mojavelinux** [comment]
I think the alt text should be escaped in the converter instead of the parser. That's where we can handle this additional escape.

**Comment by lodestone** [comment]
@mojavelinux I worked on this today but I thought about it and wondered what the "proper" fix is.

... Given ...

```asciidoc
image::/assets/cat.png[This is a "Cat"]
```

\--is it ever valid HTML to replace the inside double quotes with single quotes, or the outer double quotes with single quotes?

```html
<img src="/assets/cat.png" alt="This is a \"Cat\""/>
```

I don't think it is.

Or is the "proper" fix to use an HTML entity:

```ruby
node.attr('alt').gsub('"',"&quot;")
```

???

**Comment by mojavelinux** [comment]
The correct way is the HTML entity, &quot.

I'd like to move all the specialchars replacements for the alt text into the converter. Otherwise, we lose information about what the original text was. I think we should add a new method to Substitutors named `sub_specialcharsattr` (or `sub_attrspecialchars`) that basically does what `sub_specialchars` does, except it also handles double quote.

See https://github.com/asciidoctor/asciidoctor/blob/master/lib/asciidoctor/substitutors.rb#L346-L362.

We need to use a separate method because any logic in sub_specialchars costs us in terms of performance.

**Comment by mojavelinux** [comment]
Btw, this is why a lot of escapers / encoders escape single quotes and double quotes. They are not required for text, but they are required in attribute values. But it's a waste to do these escapes in text because they simply aren't needed. That happens to be why we can use escape_utils to do this substitution.

**Comment by mojavelinux** [comment]
I discovered that the String#encode method in Ruby can now do this escape. However, what I discovered is that it is no faster than gsub (probably because it's doing other operations as well).

See https://ruby-doc.org/core-2.4.0/String.html#method-i-encode
